File: 1.Two Sum/Jimmy Xiang/TwoSum.py
#!/usr/bin/env python

class Solution(object):
	def twoSum(self, nums, target):
		d = {}
		for i in range(len(nums)):
			if d.get(target - nums[i], None) == None:      #not found 
				d[nums[i]] = i
			else:
				return [d[target - nums[i] ] + 1 , i + 1]  #rtype list 



nums = [6,1,2,7]
target = 9

s = Solution()
result = s.twoSum(nums, target)
print result

File: 001.Two Sum/Jimmy Xiang/TwoSum.go
package main

import (
	"fmt"
)

func twoSum(arr []int, value int) (result [2]int) {

	var res [2]int = [2]int{0, 0}

	for i, v := range arr {

		for j, o := range arr {

			if i < j && (value-v) == o {

				res[0] = i
				res[1] = j

				return res
			}
		}
	}

	return res

}

func main() {

	arr := []int{222, 333, 11, 2, 3, 0, -2, 1, 2, 3}

	var sum = 5

	result := twoSum(arr, sum)

	fmt.Printf("arr[%d]:%d  + arr[%d]:%d  =  %d \n", result[0], arr[result[0]], result[1], arr[result[1]], sum)

}
